hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/AbstractDsAssignmentStrategy.java

```diff
@@ -196,9 +201,19 @@ protected List<Long> closeObsoleteUpdateActions(final Collection<Long> targetsId
      * @param actionId
      *            the action id of the assignment
      */
-    void cancelAssignDistributionSetEvent(final Target target, final Long actionId) {
+    void cancelAssignDistributionSetEvent(final Action action) {
         afterCommit.afterCommit(() -> eventPublisherHolder.getEventPublisher().publishEvent(
-                new CancelTargetAssignmentEvent(target, actionId, eventPublisherHolder.getApplicationId())));
+              new CancelTargetAssignmentEvent(action, eventPublisherHolder.getApplicationId())));
+    }
+
+    void cancelAssignDistributionSetEvent(final List<Action> actions) {
+        if (actions.isEmpty()) {
```

Comment:
please use CollectionUtils.isEmpty instead to prevent NullPointer exception

Reply:
used